### PR TITLE
DTSPO-22 User assigned managed identity

### PR DESCRIPTION
**Pull request by timja**
Requires https://github.com/hmcts/aks-module-kubernetes/pull/10

### CODE REVIEW

File: config/terraform/ss/vars/04-aks01/demo.tfvars [hunk 1]
```diff
@@ -4,5 +4,5 @@ kubernetes_cluster_version = "1.19.3"
 
 kubernetes_cluster_agent_min_count = "1"
 kubernetes_cluster_agent_max_count = "3"
-kubernetes_cluster_agent_vm_size   = "Standard_D4s_v3"
+kubernetes_cluster_agent_vm_size   = "Standard_DS3_v2"
```
[the remark this answers is not in the document]
Comment (timja) [reply config/terraform/ss/vars/04-aks01/demo.tfvars:7]: I'll remove the overrides in a separate PR

File: terraform/aks/04-aks01/20-aks.tf [hunk 2]
```diff
@@ -10,20 +10,21 @@ resource "azurerm_resource_group" "kubernetes_resource_group" {
 }
 
 module "loganalytics" {
-  source = "../../../modules/loganalytics"
+  source      = "git::https://github.com/hmcts/terraform-module-log-analytics-workspace-id.git?ref=master"
   environment = var.environment
-
 }
 
 module "kubernetes" {
```
Comment (naikajah) [inline terraform/aks/04-aks01/20-aks.tf:17]: Why are there 2 different folders for aks-00 and aks-01?

Comment (timja) [reply terraform/aks/04-aks01/20-aks.tf:17]: inherited it like that, they are identical except for the variable `cluster_number = "01"`

(should be cleaned up as tech debt)

Comment (naikajah) [reply terraform/aks/04-aks01/20-aks.tf:17]: making sure that the tech debt should not be carried forward for a long time. Addition of more constructs/features might make it difficult to restructure in future.

Comment (timja) [reply terraform/aks/04-aks01/20-aks.tf:17]: there's mean a large amount of tech debt on this repo we've fixed... including lots during this story, we're paying it down as we go

Comment (naikajah) [reply terraform/aks/04-aks01/20-aks.tf:17]: That's really good. I am only pointing out the issues/risk in doing future refactoring/resolving tech debt for the code. As you rightly said that you guys have already fixed a large amount of tech debt. Its always good to fix the tech-debts before someone points out "a large amount of tech debts"

File: pipeline-steps/deploy-service.yaml [hunk 3]
```diff
@@ -10,7 +10,7 @@ steps:
 
   - template: keyvault-read.yaml
     parameters:
-      serviceConnection: $(serviceConnection)
+      serviceConnection: $(gaServiceConnection)
```
Comment (timja) [inline pipeline-steps/deploy-service.yaml:13]: I will do more cleanup in a follow up PR for this

**Review by adusumillipraveen — APPROVED** [review]

**Pull request merged**